Subject: [PATCH] vcpkg-based dependency management.
---
Index: extlibs/module-ports/vk_mem_alloc.cppm
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/extlibs/module-ports/vk_mem_alloc.cppm b/extlibs/module-ports/vk_mem_alloc.cppm
--- a/extlibs/module-ports/vk_mem_alloc.cppm	(revision 71df52ccbeb941f4d1906106164f42305ecae388)
+++ b/extlibs/module-ports/vk_mem_alloc.cppm	(date 1721814762147)
@@ -1,6 +1,6 @@
 module;
 #define VMA_IMPLEMENTATION
-#include <vk_mem_alloc.hpp>
+#include <vulkan-memory-allocator-hpp/vk_mem_alloc.hpp>
 export module vk_mem_alloc_hpp;
 
 export namespace VMA_HPP_NAMESPACE {
Index: cmake/config.cmake.in
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cmake/config.cmake.in b/cmake/config.cmake.in
--- a/cmake/config.cmake.in	(revision 71df52ccbeb941f4d1906106164f42305ecae388)
+++ b/cmake/config.cmake.in	(date 1721814732167)
@@ -8,6 +8,6 @@
     find_dependency(Vulkan COMPONENTS shaderc_combined)
 endif()
 find_dependency(VulkanMemoryAllocator)
-find_dependency(VulkanMemoryAllocator-Hpp)
+find_dependency(unofficial-vulkan-memory-allocator-hpp)
 
 check_required_components(@PROJECT_NAME@)
\ No newline at end of file
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision 71df52ccbeb941f4d1906106164f42305ecae388)
+++ b/CMakeLists.txt	(date 1721814762150)
@@ -37,7 +37,7 @@
 
 find_package(Vulkan 1.3.256 REQUIRED)
 find_package(VulkanMemoryAllocator CONFIG REQUIRED)
-find_package(VulkanMemoryAllocator-Hpp CONFIG REQUIRED)
+find_package(unofficial-vulkan-memory-allocator-hpp CONFIG REQUIRED)
 
 if (VKU_USE_SHADERC)
     find_package(Vulkan COMPONENTS shaderc_combined REQUIRED)
@@ -96,7 +96,7 @@
 target_link_libraries(vku PUBLIC
     Vulkan::Vulkan
     GPUOpen::VulkanMemoryAllocator
-    VulkanMemoryAllocator-Hpp::VulkanMemoryAllocator-Hpp
+    unofficial::VulkanMemoryAllocator-Hpp::VulkanMemoryAllocator-Hpp
     $<$<BOOL:${VKU_USE_SHADERC}>:Vulkan::shaderc_combined>
 )
 target_compile_definitions(vku PUBLIC
